### PR TITLE
TF-2925 Store previous mailboxId to prevent upload ThreadView when change after do any action

### PR DESCRIPTION
**Pull request by hoangdat**

### CODE REVIEW

**Review by tddang-linagora — APPROVED** [review]
lgtm

File: lib/features/thread/presentation/thread_controller.dart [hunk 1]
```diff
@@ -235,10 +234,17 @@ class ThreadController extends BaseController with EmailActionController {
...
     ever(mailboxDashBoardController.selectedMailbox, (mailbox) {
-      log('ThreadController::_registerObxStreamListener:ever: SELECTED_MAILBOX_ID = ${mailbox?.id.asString} | SELECTED_MAILBOX_NAME = ${mailbox?.name?.name}');
-      _resetToOriginalValue();
-      if (mailbox != null) {
-        _getAllEmailAction();
+      log('ThreadController::_registerObxStreamListener: selectedMailbox: ${mailbox?.id} - ${mailbox?.name}');
+      log('ThreadController::_registerObxStreamListener: previousSelectedMailboxId: ${mailboxDashBoardController.previousSelectedMailboxId.value}');
+      if (mailbox is PresentationMailbox) {
+        if (mailbox.mailboxId != mailboxDashBoardController.previousSelectedMailboxId.value) {
+          _currentMailboxId = mailbox.id;
```
Comment (dab246) [inline lib/features/thread/presentation/thread_controller.dart:241]: IMO, we do not need `previousSelectedMailboxId`.

File: lib/features/thread/presentation/thread_view.dart [hunk 2]
```diff
@@ -654,17 +667,26 @@ class ThreadView extends GetWidget<ThreadController>
...
+              iconSVG: controller.imagePaths.icEmptyEmail,
+              subTitle: _getSubMessageEmptyEmail(context),
+              onCreateFiltersActionCallback: controller.isNewFolderCreated
+                ? controller.goToCreateEmailRuleView
+                : null,
+            );
+          }
+        } else {
+          return const SizedBox.shrink();
+        }
```
Comment (tddang-linagora) [inline lib/features/thread/presentation/thread_view.dart:688]:
```suggestion
        if (success is GetAllEmailLoading || success is SearchingState) {
          return const SizedBox.shrink();
        }
        
        if (success is GetAllEmailSuccess
            && success.currentMailboxId != controller.selectedMailboxId) {
          return const SizedBox.shrink();
        } else {
          return EmptyEmailsWidget(
            key: const Key('empty_thread_view'),
            title: _getMessageEmptyEmail(context),
            iconSVG: controller.imagePaths.icEmptyEmail,
            subTitle: _getSubMessageEmptyEmail(context),
            onCreateFiltersActionCallback: controller.isNewFolderCreated
              ? controller.goToCreateEmailRuleView
              : null,
          );
        }
```

Avoid nesting `if` statements

Comment (dab246) [reply lib/features/thread/presentation/thread_view.dart]: done

File: lib/features/thread/presentation/thread_view.dart [hunk 3]
```diff
@@ -247,13 +248,23 @@ class ThreadView extends GetWidget<ThreadController>
...
+      final isSearchEmailRunning = controller.searchController.simpleSearchIsActivated.isTrue
+          || controller.searchController.advancedSearchIsActivated.isTrue;
+
+      if ((PlatformInfo.isMobile
+            || (PlatformInfo.isWeb
+              && currentSelectMode == SelectMode.ACTIVE
+              && isSearchEmailRunning
+              && !controller.responsiveUtils.isDesktop(context)
+            ))
+          && listEmailSelected.isNotEmpty
```
[the remark this answers is not in the document]
Comment (dab246) [reply lib/features/thread/presentation/thread_view.dart]: done

**Review by tddang-linagora — APPROVED** [review]

**Comment by hoangdat** [comment]
Approved. Please rebase

**Comment by dab246** [comment]
> Approved. Please rebase

Done

**Review by dab246 — APPROVED** [review]
LGTM

**Pull request merged**